spock-specs/src/test/groovy/org/spockframework/smoke/extension/TempDirExtensionSpec.groovy

```diff
@@ -0,0 +1,104 @@
+package org.spockframework.smoke.extension
+
+import org.spockframework.util.Beta
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Stepwise
+import spock.lang.TempDir
+
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.attribute.PosixFilePermissions
+
+/**
+ * @author dqyuan
+ */
+@Beta
+@Stepwise
+class TempDirExtensionSpec extends Specification {
+
+  @Shared
+  @TempDir
+  Path sharedDir
+  static Path preShared
+
+  @TempDir
+  File iterationDir
+  static File preIteration
+
+  @TempDir(baseDir = "build")
+  Path customerTempDir
+
+  @TempDir(baseDir = "build", reserveAfterTest = { true })
+  Path reserveDir
+  static Path reserveFile
+
+  def "temp dir exist"() {
+    preShared = sharedDir
+    preIteration = iterationDir
+    reserveFile = reserveDir.resolve("test.txt")
+    Files.write(reserveFile, "aaa".getBytes())
+    expect:
+    assert Files.exists(sharedDir)
+    assert Files.exists(Paths.get(iterationDir.toURI()))
```

Comment:
Any reason not to simply do `iterationDir.exists()` or even better `iterationDir.directory` which checks that it exists and that it is a directory?